### PR TITLE
WW-28

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: src/main/resources/FileHandleRecord.ddl.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE IF NOT EXISTS `FILE_HANDLE_RECORD` (
+  `ID` bigint NOT NULL,
+  `CREATED_ON` bigint NOT NULL,
+  `CREATED_BY` bigint NOT NULL,
+  `CONCRETE_TYPE` varchar(100) CHARACTER SET latin1 COLLATE latin1_bin NOT NULL,
+  `CONTENT_SIZE` bigint DEFAULT NULL,
+  `BUCKET` varchar(100) CHARACTER SET latin1 COLLATE latin1_bin DEFAULT NULL,
+  `_KEY` varchar(100) CHARACTER SET latin1 COLLATE latin1_bin DEFAULT NULL,
```
Comment (john-hill) [inline src/main/resources/FileHandleRecord.ddl.sql]: KEY_OR_PATH

Comment (john-hill) [reply src/main/resources/FileHandleRecord.ddl.sql]: 700

File: src/main/resources/FileHandleRecord.ddl.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  `FILE_NAME` varchar(100) CHARACTER SET latin1 COLLATE latin1_bin NOT NULL,
```
Comment (john-hill) [inline src/main/resources/FileHandleRecord.ddl.sql]: 256

File: src/test/java/org/sagebionetworks/warehouse/workers/db/snapshot/FileHandleRecordDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		assertEquals(record1, dao.get(Long.parseLong(record1.getId())));
+		assertEquals(record2, dao.get(Long.parseLong(record2.getId())));
+
+		FileHandleSnapshot record3 = ObjectSnapshotTestUtil.createValidFileHandleSnapshot();
+		record3.setId(record2.getId());
+		record3.setFileName("new file name");
+		dao.insert(Arrays.asList(record3));
+		assertEquals(record2, dao.get(Long.parseLong(record3.getId())));
+	}
+
```
Comment (john-hill) [inline src/test/java/org/sagebionetworks/warehouse/workers/db/snapshot/FileHandleRecordDaoImplTest.java:48]: add test for insert with non-null values for all optional fields.
